web/src/main/resources/static/app.js

```diff
@@ -132,78 +125,50 @@ class Vehicle {
     this._lastKnownPosition = position;
   }
 
-  updateData({position, schedule}) {
+  updateData({ position, schedule }) {
     this.schedule = schedule;
     this._lastKnownPosition = position;
 
     if (this._hasMovementEnded) {
-      DEBUG && console.log(555, 'xxx movement has ended, removing.', this.vehicleId)
+      DEBUG && console.log(`movement for vehicle ${this.vehicleId} has ended, removing.`)
       this._onFinalStop();
       return;
     }
 
-    // xxx why the time diff?
-    if (this._lastKnownPosition.speed = 0) return
-    // if (!this._hasMovementStarted) {
-    //   DEBUG && console.log(555, 'xxx movement not started, not doing anything.', this.vehicleId)
-    //   DEBUG && console.log(555, currentTime(), this.schedule[0].arrival, this._lastKnownPosition.speed)
-    //   return;
-    // }
-
     if (!this._marker) {
       this._createMarker();
       this._route.setColor();
     }
-
-    // this._debugMarker(this._lastKnownLatLong, `${this.vehicleId} - ${new Date()}`)
-
-    // this._route.latlngs.forEach(latlong => {
-    //   console.log(999, latlong.distanceTo(this._marker.getLatLng()))
-    // })
-
-    // const distancesToMarker = this._route.latlngs.map(latlong => latlong.distanceTo(this._marker.getLatLng()))
-    // const closestDistance = Math.min(...distancesToMarker)
-    // const idx = distancesToMarker.findIndex(distance => distance === closestDistance)
-
-    // console.log(999999, `closest stop: (${idx}/${this._route.latlngs.length})`)
-    // const nextStopIdx = this.schedule.findIndex(({ latitude, longitude }) => t < arrival);
   }
 
   _createMarker() {
     const icon = ROUTE_ICON_MAPPING[this.routeType]
 
-    if (OPTIONS.VEHICLE_COLORS == 'FUNKY') {
-      icon.options.html = icon.options.html.replace(/style="[^\"]*"/g, `style="color: ${randomColor()}"`)
+    if (OPTIONS.VEHICLE_COLORS === 'FUNKY') {
```

Comment:
This morning, I create a PR #43 to use the routes' color. In that case, would you please:

1. review it
2. accept the merge if it's OK
3. rebase this branch on top
4. remove these checks because colors are set in the file